### ISSUE TITLE
Circular dependency issue

### ISSUE THREAD

**Issue by neelasha23** [ploomber/jupysql#545]
There's a circular dependency whenever we import `util.py` in other modules like `store.py`, etc.

```
ImportError while loading conftest '/jupysql/src/tests/conftest.py'.
src/tests/conftest.py:8: in <module>
    from sql.magic import SqlMagic, RenderMagic
../../miniforge3/envs/jupysql/lib/python3.10/site-packages/sql/__init__.py:1: in <module>
    from .magic import RenderMagic, SqlMagic, load_ipython_extension
../../miniforge3/envs/jupysql/lib/python3.10/site-packages/sql/magic.py:22: in <module>
    import sql.connection
../../miniforge3/envs/jupysql/lib/python3.10/site-packages/sql/connection.py:11: in <module>
    from sql.store import store
../../miniforge3/envs/jupysql/lib/python3.10/site-packages/sql/store.py:10: in <module>
    from sql import util
../../miniforge3/envs/jupysql/lib/python3.10/site-packages/sql/util.py:3: in <module>
    from sql import inspect
../../miniforge3/envs/jupysql/lib/python3.10/site-packages/sql/inspect.py:4: in <module>
    from sql.connection import Connection
E   ImportError: cannot import name 'Connection' from partially initialized module 'sql.connection' (most likely due to a circular import) (/Users/neelashasen/miniforge3/envs/jupysql/lib/python3.10/site-packages/sql/connection.py)
```

**Comment by edublancas** [comment]
the cleanest way to fix this is to clean up `util.py`, so it doesn't have that many imports: it currently imports inspect, Connection, store. This makes `util.py` "difficult" to import because importing it in any of its existing imports creates the circular dependency.

so we should remove most of those imports and move that logic to a different file. essentially, we need to move out the functionality in `util.py` that depends on an active connection to `inspect.py`

**Comment by edublancas** [comment]
fixed by #787
